shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/plugins/PluginManager.java

```diff
@@ -1548,20 +1550,26 @@ protected void newInstance(GeneratorAdapter method) {
     }
 
     /**
-     * Perform validation on methods annotated with @ShesmuInputSource and track as either a "static" or "dynamic"
-     * type of source.
-     * Validation on a @ShesmuInputSource method is as follows:
-     *  1. Ensure method returns Stream
-     *  2. Ensure method returns Stream<T> or a subclass of T but not a wildcard
-     *  3. Ensure method has either 0 parameters, or 1 parameter of type boolean
-     *     (this is the optional readStale boolean)
-     * If the ShesmuInputSource is from an instance source (i.e., a virtual method in a class that extends PluginFile)
-     * then the method is tracked as a dynamic source. These input sources require a configuration file.
-     * If the ShesmuInputSource is from a repository source (i.e., a static method in a class that extends
-     * PluginFileType) then the method is tracked as a static source. These input sources require no configuration.
-     * See implementation.md
+     * Perform validation on methods annotated with {@link ShesmuInputSource} and track as either a
+     * "static" or "dynamic" type of source. Validation on a {@link ShesmuInputSource} method is as
+     * follows:
      *
-     * @param method Method object of @ShesmuInputSource-annotated method to process
+     * <ol>
+     *   <li>Ensure method returns Stream
+     *   <li>Ensure method returns {@link Stream} or a subclass of the type parameter but not a
```

Comment:
"Stream matching the type parameter?" Probably helpful to mention the location of the type parameter too (`FormatTypeWrapper`)?

Reply:
I've tried to clarify this. I'm not sure what you mean about `FormatTypeWrapper` as it's not used here.

Reply:
The method checks whether the method return type is `instanceof ParameterizedType`, and if not throws an exception with message

> Method %s of %s does not return Stream<T> or generic type information missing from bytecode.

and the docs originally referred to `Stream<T>` and `T`. Since the method is in the `FormatTypeWrapper` class, which has a type parameter `T`, I assumed this was the type parameter being referred to, but it seems that the method doesn't use any `T` at all, so unless I'm missing something, the exception message is wrong, but the updated JavaDoc is correct (and much clearer now, thanks!)

Reply:
The exception isn't very helpful. The `T` is not referring to any `T` in this class. The issue here is how Java encodes generics... It's possible to write a method `Stream foo()` in a plugin which has no generic type information, and that's a problem (the one detected by line 1589). Maybe the exception should read `does not return Stream<> or generic type information missing`? Or maybe `Stream<...>`?

Reply:
Ah, makes sense. I think it is slightly clearer without the `T`